### PR TITLE
Improvements

### PR DESCRIPTION
**Pull request by SergeyTsimfer**
This PR:

- drastically improves memory footprint of `Controller`s

- uses a better naming system for horizons in `Controller`s

- makes scripts work better with boolean values

### CODE REVIEW

File: seismiqb/src/controllers/extender.py [hunk 1]
```diff
@@ -81,13 +83,19 @@ def inference(self, horizon, n_steps=30, batch_size=128, stride=16):
...
             curr_len = len(horizon)
             if (curr_len - prev_len) < 25:
                 break
             self.log(f'Extended from {prev_len} to {curr_len}, + {curr_len - prev_len}')
             prev_len = curr_len
 
+            # Cleanup
+            inference_pipeline.reset('variables')
+            inference_pipeline = None
+
```
Comment (dmylzenova) [inline seismiqb/src/controllers/extender.py:96]: gc.collect?

File: seismiqb/src/horizon.py [hunk 2]
```diff
@@ -931,11 +931,11 @@ def get_cube_values(self, window=23, offset=0, scale=False, chunk_size=256):
...
             scale = lambda array: array
 
-        for h_start in range(self.h_min - low, self.h_max + high, chunk_size):
-            h_end = min(h_start + chunk_size, self.h_max + high, self.geometry.depth)
+        for h_start in range(max(low, self.h_min), self.h_max, chunk_size):
+            h_end = min(h_start + chunk_size, self.h_max + 1)
 
             # Get chunk from the cube (depth-wise)
-            data_chunk = cube_hdf5[h_start:h_end, :, :]
+            data_chunk = cube_hdf5[(h_start-low) : min(h_end+high, self.geometry.depth), :, :]
```
Comment (dmylzenova) [inline seismiqb/src/horizon.py:938]: Don't you know the min and max horizon depth? Why do you need dealing with all ranges beforehand?

File: seismiqb/src/controllers/base.py [hunk 3]
```diff
@@ -157,7 +159,7 @@ def make_dataset_from_horizon(self, horizon):
 
         dsi = FilesIndex(path=[cube_path], no_ext=True)
         dataset = SeismicCubeset(dsi)
-        dataset.load_geometries()
+        dataset.geometries[dataset.indices[0]] = horizon.geometry
```
Comment (dmylzenova) [inline seismiqb/src/controllers/base.py:162]: 👍

File: seismiqb/src/controllers/base.py [hunk 4]
```diff
@@ -102,8 +104,9 @@ def make_logger(self):
     def log(self, msg):
         """ Log supplied message. """
         if self.logger is not None:
-            self.logger(f'{self.__class__.__name__} ::: {msg}')
-
+            process = psutil.Process(os.getpid())
+            uss = process.memory_full_info().uss / (1024 ** 3)
+            self.logger(f'{self.__class__.__name__} ::: {uss:2.4} ::: {msg}')
```
Comment (dmylzenova) [inline seismiqb/src/controllers/base.py:109]: This is cool!

File: seismiqb/src/controllers/base.py [hunk 5]
```diff
@@ -452,6 +468,13 @@ def inference_1(self, dataset, heights_range=None, orientation='i', overlap_fact
                                                dataset.grid_info, threshold=0.0, minsize=50)
             horizons.extend(chunk_horizons)
 
+            # Cleanup
+            inference_pipeline.reset('variables')
```
Comment (akoryagin) [inline seismiqb/src/controllers/base.py:472]: did it help?

**Pull request merged**